src/schematools/utils.py

```diff
@@ -188,7 +187,7 @@ def dataset_schema_from_file(
 
 def dataset_schema_from_path(
     dataset_path: Union[Path, str],
-) -> types.DatasetSchema:
+) -> types.ST:
```

Comment:
dit moet toch gewoon `DaasetSchema` zijn?
`types.ST` is een generic type, raar dat mypy(?) dat voorstelde. Je gebruikt hier `DatasetSchema.from_dict()` en niet de base-class versie waarbij dat ontbreekt. Ik denk dat die `SchemaType.from_dict()` zelfs verwijderd mag worden.

Reply:
Ik zie op nog een paar plekken dat de concrete class gebruikt moet worden, ipv. die generieke, ga ik nog ff naar kijken.